esi_report.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# run ./esi_report.sh daily to get daily report
+# run ./esi_report.sh monthly to get last month's report
+# run ./esi_report.sh weekly to get last week's report
+# run ./esi_reports.sh plan to get future plan
+
+report_file=/tmp/esi_report
+
+source /home/stack/overcloud-deploy/overcloud/overcloudrc
+
+write_output () {
+    openstack esi lease list -f value |
+    while read line
+    do
+        start_time=$(cut -d' ' -f5 <<<$line)
+        end_time=$(cut -d' ' -f6 <<<$line)
+        if [[ $end_time < $first_day || $start_time > $last_day ]]; then
+            :
+        else
+            echo $line >> $output_file
+        fi
+    done
+}
+
+if [[ $1 == "weekly" ]]; then
+    week=$(date -d "last Sunday -6 days" +"%Y-%m-%d")
+    output_file=${report_file}_weekly.log.${week}
+    first_day=$(date -d "last Sunday -6 days" +"%Y-%m-%dT%H:%M:%S")
+    last_day=$(date -d "last Sunday 23:59:59" +"%Y-%m-%dT%H:%M:%S")
+    echo "===== Start generating weekly report for week ${week} =====" >> $output_file
+    echo "UUID Resource Resource_Class Project Start_Time End_Time Offer_UUID Status" >> $output_file
+    write_output
```

Comment:
If a single-line esi command works, then we can just use that here (and in the monthly report).